include/solvers/mpm_explicit.tcc

```diff
@@ -170,11 +170,15 @@ bool mpm::MPMExplicit<Tdim>::solve() {
 
     task_group.wait();
 
-    // Assign material ids to node
-    if (interface_)
+    // Initialise nodal properties and assign material ids to node
+    if (interface_) {
+      // Initialise nodal properties
+      mesh_->initialise_properties();
+
       mesh_->iterate_over_particles(
```

Comment:
Does this reset the material ids already present in the node?

Reply:
It does not. It only resets the nodal properties' values. But now that you have said it, it should also reset the material ids in the node as well, right?

Reply:
material_ids_.clear() is called in the Node class (`initialise()` function), thus already being reset as the nodes are reset.